source/as-promise/core.ts

```diff
@@ -0,0 +1,139 @@
+import {URL} from 'url';
+import is, {assert} from '@sindresorhus/is';
+import {
+	Options,
+	NormalizedOptions,
+	Defaults,
+	ResponseType
+} from './types';
+import Request, {knownHookEvents, RequestError, HTTPError} from '../core';
+
+if (!knownHookEvents.includes('beforeRetry' as any)) {
+	knownHookEvents.push('beforeRetry' as any, 'afterResponse' as any);
```

Comment:
The normalizing will be done via `Request.normalizeArguments` so we just tell `Request` that it needs to normalize `beforeRetry` too.